tests/borrow/typeclassborrow.nim

```diff
@@ -0,0 +1,32 @@
+type
```

Comment:
A few one liners saying the intention of the various cases is the only thing that takes some guess work to figure out.